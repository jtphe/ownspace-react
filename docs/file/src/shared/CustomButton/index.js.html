<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/shared/CustomButton/index.js</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
</head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
</header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#api">api</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-api">api</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-document">components/Document</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-BackFolder">BackFolder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Folder">Folder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-UploadDocument">UploadDocument</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-home">components/Home</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CreateFile">CreateFile</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Home">Home</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-login">components/Login</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-GenerateTotp">GenerateTotp</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Login">Login</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-SecondAuthScreen">SecondAuthScreen</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TotpAuthScreen">TotpAuthScreen</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-menu">components/Menu</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-IconMenu">IconMenu</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-NavMenu">NavMenu</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-userprofile">components/UserProfile</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-PasswordModal">PasswordModal</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#constants">constants</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-RECURRING_ROLE">RECURRING_ROLE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_ONE">ROLE_ONE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_THREE">ROLE_THREE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_TWO">ROLE_TWO</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#router">router</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-AppRouter">AppRouter</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-avatar">shared/Avatar</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Avatar">Avatar</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-clienttext">shared/ClientText</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextClient">TextClient</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-custombutton">shared/CustomButton</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CustomButton">CustomButton</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-dialog">shared/Dialog</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OwnSpaceDialog">OwnSpaceDialog</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-header">shared/Header</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Header">Header</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-WithBack">WithBack</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-logo">shared/Logo</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Logo">Logo</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-text">shared/Text</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextHelvetica">TextHelvetica</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-openFile">openFile</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-requestAndroidWriteExternalStorage">requestAndroidWriteExternalStorage</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-octetToMoConverter">octetToMoConverter</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-formatDate">formatDate</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-decryptPassword">decryptPassword</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-hashPassword">hashPassword</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-showToast">showToast</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/shared/CustomButton/index.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import React from &apos;react&apos;;
import { View, StyleSheet } from &apos;react-native&apos;;
import { Button } from &apos;react-native-paper&apos;;
import { Actions } from &apos;react-native-router-flux&apos;;
import i18n from &apos;@i18n/i18n&apos;;
import { useFonts } from &apos;@use-expo/font&apos;;

/**
 * The CustomButton component
 * @param {function} confirmFunction - Confirm the function passed to the confirm button
 */
const CustomButton = ({ confirmFunction }) =&gt; {
  useFonts({
    // eslint-disable-next-line global-require
    DejaVuSans: require(&apos;../../../assets/fonts/DejaVuSans.ttf&apos;)
  });

  /**
   * Render the CustomButton component
   * @returns {React.Component} - CustomButton component
   */
  return (
    &lt;View style={styles.btnContainer}&gt;
      &lt;Button
        mode=&quot;contained&quot;
        uppercase={false}
        labelStyle={styles.cancelText}
        contentStyle={styles.btnDimension}
        style={styles.btnCancel}
        onPress={() =&gt; Actions.pop()}
      &gt;
        {i18n.t(&apos;button.cancel&apos;)}
      &lt;/Button&gt;
      &lt;Button
        mode=&quot;contained&quot;
        uppercase={false}
        labelStyle={styles.confirmText}
        contentStyle={styles.btnDimension}
        style={styles.btnConfirm}
        onPress={() =&gt; confirmFunction()}
      &gt;
        {i18n.t(&apos;button.validate&apos;)}
      &lt;/Button&gt;
    &lt;/View&gt;
  );
};

/**
 * Styles of CustomButton component
 */
const styles = StyleSheet.create({
  btnContainer: {
    flexDirection: &apos;row&apos;,
    padding: 35,
    paddingTop: 20,
    paddingBottom: 20,
    marginBottom: 10,
    justifyContent: &apos;space-between&apos;
  },
  btnCancel: {
    justifyContent: &apos;center&apos;,
    backgroundColor: &apos;white&apos;,
    borderColor: &apos;#003466&apos;,
    borderWidth: 1,
    color: &apos;#42688B&apos;
  },
  btnConfirm: {
    justifyContent: &apos;center&apos;,
    backgroundColor: &apos;#003466&apos;,
    borderColor: &apos;black&apos;
  },
  btnDimension: {
    height: 45,
    width: 100
  },
  cancelText: { fontFamily: &apos;DejaVuSans&apos;, color: &apos;#42688B&apos; },
  confirmText: { fontFamily: &apos;DejaVuSans&apos;, color: &apos;#fff&apos; }
});

export default CustomButton;
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
