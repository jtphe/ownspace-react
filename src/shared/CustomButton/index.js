import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Button } from 'react-native-paper'
import { Actions } from 'react-native-router-flux';
import i18n from '@i18n/i18n';

const CustomButton = ({ confirmFunction }) => {
    return (
        <View style={styles.btnContainer}>
            <Button
                mode={'contained'}
                uppercase={false}
                labelStyle={{ color: '#42688B' }}
                contentStyle={styles.btnDimension}
                style={styles.btnCancel}
                onPress={() => Actions.pop()}
            >
                {i18n.t('button.cancel')}
            </Button>
            <Button
                mode={'contained'}
                uppercase={false}
                labelStyle={{ color: '#fff' }}
                contentStyle={styles.btnDimension}
                style={styles.btnConfirm}
                onPress={() => confirmFunction()}
            >
                {i18n.t('button.validate')}
            </Button>
        </View>
    )
}

const styles = StyleSheet.create({
    btnContainer: { flexDirection: 'row', padding: 35, paddingTop: 20, paddingBottom: 20, marginBottom: 10, justifyContent: 'space-between', },
    btnCancel: {
        justifyContent: 'center',
        backgroundColor: 'white',
        borderColor: '#42688B',
        borderWidth: 1,
        color: '#42688B'
    },
    btnConfirm: {
        justifyContent: 'center',
        backgroundColor: '#42688B',
        borderColor: 'black'
    },
    btnDimension: {
        height: 45,
        width: 100
    }
})

export default CustomButton;