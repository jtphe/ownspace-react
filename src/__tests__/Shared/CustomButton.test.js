import 'react-native';
import React from 'react';
import CustomButton from '@shared/CustomButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from 'react-native-paper';
import { CLIENT_COLOR_PRIMARY } from '@constants';

describe('CustomButton tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CustomButton />);
  });

  it('CustomButton snapshot', () => {
    const tree = renderer.create(<CustomButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Test buttons style', () => {
    const cancelBtn = wrapper.find(Button).get(0);
    const validateBtn = wrapper.find(Button).get(1);

    expect(cancelBtn.props.labelStyle).toEqual({
      fontFamily: 'DejaVuSans',
      color: CLIENT_COLOR_PRIMARY
    });
    expect(validateBtn.props.labelStyle).toEqual({
      fontFamily: 'DejaVuSans',
      color: '#fff'
    });
  });
});

Enzyme.configure({ adapter: new Adapter() });
