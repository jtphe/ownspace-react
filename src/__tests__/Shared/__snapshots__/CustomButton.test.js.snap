// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton tests CustomButton snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "padding": 35,
      "paddingBottom": 20,
      "paddingTop": 20,
      "zIndex": -1,
    }
  }
>
  <View
    id="cancelBtn"
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#003466",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#003466",
        "elevation": 2,
        "justifyContent": "center",
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 45,
              "width": 100,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                false,
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#003466",
                  "fontFamily": "DejaVuSans",
                },
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#003466",
        "borderColor": "black",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "justifyContent": "center",
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 45,
              "width": 100,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                false,
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "DejaVuSans",
                },
              ],
            ]
          }
        >
          Validate
        </Text>
      </View>
    </View>
  </View>
</View>
`;
